pandas/_libs/lib.pyx

```diff
@@ -1203,9 +1204,14 @@ cdef object _try_infer_map(object dtype):
         object val
         str attr
     for attr in ["name", "kind", "base"]:
-        val = getattr(dtype, attr)
+        val = getattr(dtype, attr, None)
```

Comment:
running into this in a branch implementing ExtensionIndex for general EAs: should we require a `base` attribute?  In particular JSONDtype and DecimalDtype break here

Reply:
I don't think we should require a `base` attribute. What does it actually mean? AFAIK it is only useful for dtypes that describe sub-arrays, which I have not yet ever used

Reply:
As long as `lib.infer_dtype(decimal_array)` isn't raising an AttributeError, i'll be a happy camper.